src/main/java/sirius/biz/tycho/search/OpenSearchController.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.tycho.search;
+
+import com.alibaba.fastjson.JSONArray;
+import com.alibaba.fastjson.JSONObject;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import sirius.biz.tycho.QuickAction;
+import sirius.biz.web.BizController;
+import sirius.kernel.async.Barrier;
+import sirius.kernel.async.Future;
+import sirius.kernel.async.Tasks;
+import sirius.kernel.commons.Explain;
+import sirius.kernel.commons.Strings;
+import sirius.kernel.commons.Timeout;
+import sirius.kernel.di.PartCollection;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.di.std.Parts;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.health.Exceptions;
+import sirius.kernel.health.Log;
+import sirius.web.controller.Routed;
+import sirius.web.http.WebContext;
+import sirius.web.security.LoginRequired;
+import sirius.web.services.JSONStructuredOutput;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Contains the main parts of the OpenSearch engine for the Tycho UI.
+ * <p>
+ * The task of the OpenSearch is to search all matching entities for a given search query. As this yields quite a lot
+ * of queries, we use two thread pools. This first (<tt>tycho-open-search</tt>) limits the maximal number of parallel
+ * searches being executed. This is done in order to protect the system as each search invoked multiple providers
+ * which each execute one or more database queries. Now as some providers will be faster than others and as we want
+ * to optimize the user experience, we execute each provider in a separate thread in the second thread pool
+ * (<tt>tycho-open-search-task</tt>). Now in order for all this to make sense, we're not simply generating a plain
+ * JSON response as that way we'd had to wait for even the slowest provider to show any results. We rather use
+ * a "comet" link which sends individual search results as genrated by each provider as single JSON objects, each
+ * in its own line (separated by a <tt>\n</tt>).
+ * <p>
+ * On the client side we use the <tt>oboe</tt> library which handles this kind of incoming JSON objects and renders
+ * any incoming results as fast as possible.
+ */
+@Register
```

Comment:
Should this be bound to a framework/role ?

Reply:
As this will be part of the default UI of sirius I'd always turn it on.
If someone wants to use the URIs differently, a lower prio value could be used.

I'd say this is more convenient than having to remember to turn on all kinds of frameworks for the UI to work...